Projects/surface_photometry.ipynb

```diff
@@ -0,0 +1,1332 @@
+    "\n",
+    "If you use the techniques presented here, please cite:\n",
+    "[Watkins et al. (2016)](https://ui.adsabs.harvard.edu/abs/2016ApJ...826...59W/abstract) and\n",
+    "[Chamba et al. (2020)](https://ui.adsabs.harvard.edu/abs/2020A%26A...633L...3C/abstract)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## **Initial Setup**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# What version of the Stack am I using?\n",
+    "! echo $HOSTNAME\n",
+    "! eups list lsst_distrib -s"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### **Packages and dependencies**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Image retrieval tasks\n",
+    "from lsst.obs.hsc.hscMapper import HscMapper\n",
+    "import lsst.daf.persistence as dafPersist\n",
+    "from lsst.daf.persistence import Butler\n",
+    "import lsst.afw.table as afwTable\n",
+    "import lsst.afw.display as afwDisplay\n",
+    "import lsst.afw.detection as afwDetection\n",
+    "import lsst.geom as geom\n",
+    "from lsst.geom import Box2D\n",
+    "from lsst.afw.image import Exposure, Image, PARENT, MultibandExposure, MultibandImage\n",
+    "\n",
+    "# Detection and deblending tasks\n",
+    "from lsst.meas.algorithms import SourceDetectionTask\n",
+    "from lsst.meas.base import SingleFrameMeasurementTask\n",
+    "from lsst.afw.table import SourceCatalog\n",
+    "import scarlet\n",
+    "import scarlet.display\n",
+    "import lsst.meas.extensions.scarlet as mes\n",
+    "from scarlet_extensions.initialization import detection as det\n",
+    "\n",
+    "# All other (non-lsst) tasks\n",
+    "import os\n",
+    "import sys\n",
+    "import glob\n",
+    "from copy import copy\n",
+    "import warnings\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib.colors as colors\n",
+    "from matplotlib.patches import Ellipse\n",
+    "from IPython.display import display, Markdown\n",
+    "import numpy as np\n",
+    "from astropy.table import Table\n",
+    "import astropy.coordinates as coord\n",
+    "from astropy.visualization import make_lupton_rgb\n",
+    "\n",
+    "afwDisplay.setDefaultBackend('matplotlib')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "One final dependency is the following:\n",
+    "https://github.com/herjy/scarlet_extensions\n",
+    "\n",
+    "Git clone this, then move the scarlet_extensions subdirectory into this notebook's directory to perform the deblending aspects of this notebook.\n",
+    "\n",
+    "This package requires SEP (SExtractor for Python).  In an LSST terminal, do `pip install sep`."
```

Comment:
Is it first necessary to setup the stack tools in your terminal? I was running into an issue that without setting up lsst_distrib, the default pip was installing for python3.6 while my notebook was running python3.7.

Reply:
if you mean that we need to have activated the lsst_distrib before running the notebook and installing the external packages, then yes! But @aemerywatkins did this part, maybe he can comment?

Reply:
I admit this is a very inelegant way of doing things.  My procedure here was, in the RSP, to open up a new LSST console, then run `pip install sep`, then run the notebook.  We could just include a shell command here, actually, which would be more efficient.  Would that solve the confusion?

Reply:
Actually, I'm confused by that procedure @aemerywatkins. I would have expected that after you started the terminal you would need to have sources the LSST setup script. Otherwise, you'll be using the "system" python, which is Python3.6 instead of the LSST Python3.7 (used by the notebook). The pip install would go in the 3.6 lib path, and wouldn't be picked up by the notebook. This is based on my experience trying to replicate your setup on Friday.

Reply:
Hm... well, could be my memory is faulty.
I guess, regardless, what we need is to find a solution that will work every time.  If we just run `!pip install sep` from the notebook itself, will that install it into LSST Python 3.7?

Reply:
Ok, you actually mention later in the notebook that you need to be in an LSST env to install sep. Talking to Nushia, we figured out the issue is that the scarlet import being done at the top is forcing the user to install sep earlier than you intended
```
import lsst.meas.extensions.scarlet as mes
```